packages/datadog-esbuild/index.js

```diff
@@ -80,23 +118,26 @@ module.exports.setup = function (build) {
   })
 
   build.onLoad({ filter: /.*/, namespace: NAMESPACE }, args => {
-    if (DEBUG) {
-      console.log(`load ${args.path}@${args.pluginData.version}`)
-    }
+    const data = args.pluginData
+    const path = args.path
+
+    if (DEBUG) console.log(`LOAD ${data.pkg}@${data.version}, pkg "${path}"`)
 
-    // JSON.stringify adds double quotes. For perf gain could simply add in quotes when we know it's safe.
     const contents = `
       const dc = require('diagnostics_channel');
-      const ch = dc.channel(${JSON.stringify(DC_CHANNEL + ':' + args.path)});
-      const mod = require(${JSON.stringify(args.path)});
+      const ch = dc.channel('dd-trace-esbuild');
+      const mod = require('${path}');
       const payload = {
         module: mod,
-        path: ${JSON.stringify(args.path)},
-        version: ${JSON.stringify(args.pluginData.version)}
+        path: '${data.raw}',
+        version: '${data.version}',
+        package: '${data.pkg}',
+        relPath: '${data.path}'
```

Comment:
Is the distinction between `path` and `relPath` actually needed?

Reply:
I modified the code so that the emitter makes the decision of how to handle raw != package and now one less piece of data is being emitted.